codegenerator/cli/templates/static/codegen/src/eventFetching/chainWorkers/HyperFuelWorker.res

```diff
@@ -3,10 +3,129 @@ open Belt
 
 exception EventRoutingFailed
 
+type event = {
+  name: string,
+  logId?: string,
+  mint?: bool,
+  eventMod: module(Types.Event),
+}
+
+type contract = {
+  name: string,
+  addresses: array<Address.t>,
+  events: array<event>,
+}
+
+let makeGetRecieptsSelectionOrThrow = (~contracts: array<contract>) => {
+  let logDataReceiptTypeSelection: array<Fuel.receiptType> = [LogData]
+  let mintReceiptTypeSelection: array<Fuel.receiptType> = [Mint]
+
+  // only transactions with status 1 (success)
+  let txStatusSelection = [1]
+
+  let mint: ref<[#None | #Wildcard | #NonWildcard]> = ref(#None)
+  let nonWildcardRbsByContract = Js.Dict.empty()
+  let wildcardRbs = []
+
+  let wildcardMintSelection = (
+    {
+      receiptType: mintReceiptTypeSelection,
+      txStatus: txStatusSelection,
+    }: HyperFuelClient.QueryTypes.receiptSelection
+  )
+  let maybeWildcardLogSelection = switch wildcardRbs {
+  | [] => None
+  | wildcardRbs =>
+    Some(
+      (
+        {
+          receiptType: logDataReceiptTypeSelection,
+          txStatus: txStatusSelection,
+          rb: wildcardRbs,
+        }: HyperFuelClient.QueryTypes.receiptSelection
+      ),
+    )
+  }
+
+  contracts->Array.forEach(contract => {
+    let nonWildcardRbs = []
+    contract.events->Array.forEach(event => {
+      let module(Event) = event.eventMod
+      let {isWildcard} = Event.handlerRegister->Types.HandlerTypes.Register.getEventOptions
+      switch event {
+      | {mint: true} => mint := (isWildcard ? #Wildcard : #NonWildcard)
```

Comment:
Looks like "mint" can be over written with "#Wildcard or #NonWildcard", what should happen if both appear?